### PR TITLE
Add CoordinatorLayout and Scrolling techniques for MainActivity

### PR DESCRIPTION
**Pull request by takahirom**
Please look this.
scrolltech

This behavior is included in the Material Design Guideline.
https://www.google.com/design/spec/patterns/scrolling-techniques.html

I did the following things.
* Add CoordinatorLayout.
* Add scrolling behavior.
* Change SponsorFragment to NestedScrollView from ScrollView.
* Add padding for MapFragment. Because, MapFragment slip like this.. Bottom layout is Hidden..
image
* Disable AppBarLayout drag. Because same reason.  If can drag AppBarLayout, Bottom layout is Hidden.

I solved by setting Google Map API Key in AndroidManifest.xml.
~~I have the issue that I cannot look map...~~
~~If it is solved I remove [WIP].~~

### CODE REVIEW

**Comment by hotchemi** [comment]
@konifar thoughts?

**Comment by konifar** [comment]
:eyes:

File: app/src/main/java/io/github/droidkaigi/confsched/activity/MainActivity.java [hunk 1]
```diff
@@ -90,6 +92,29 @@ private void initView() {
         toggle.syncState();
         binding.navView.setNavigationItemSelectedListener(this);
         binding.navView.setCheckedItem(R.id.nav_all_sessions);
+
+        binding.getRoot().getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+            @Override
+            public void onGlobalLayout() {
+                if (ViewCompat.isLaidOut(binding.appbar)) {
+                    binding.getRoot().getViewTreeObserver().removeOnGlobalLayoutListener(this);
+                    initAppBarBehavior();
```
Comment (takahirom) [inline app/src/main/java/io/github/droidkaigi/confsched/activity/MainActivity.java]: If this put  `onCreate()` timing, `binding.appbar.getLayoutParams().getBehavior()` is null.
So I wait to ending layout.

File: app/src/main/java/io/github/droidkaigi/confsched/activity/MainActivity.java [hunk 2]
```diff
@@ -132,10 +156,13 @@ private void toggleToolbarElevation(boolean enable) {
         }
     }
 
-    private void changePage(@StringRes int titleRes, @NonNull Fragment fragment) {
+    private void changePage(Page page) {
+        binding.appbar.setExpanded(true, true);
```
Comment (takahirom) [inline app/src/main/java/io/github/droidkaigi/confsched/activity/MainActivity.java]: If you are in MapFragment ,You can't expand AppBarLayout. 
So I expanded AppBarLayout programmatically on changePage.

**Comment by takahirom** [comment]
I added explanation. But I'm not good at English...
If you have questions, Please tell me on twitter.

**Comment by konifar** [comment]
Thank you for your fix! But sorry. I reject this PR.

There are 2 reasons.
1. In my opinion, it is better to fix toolbar in the case that drawer exists.
2. I guess `CoordinatorLayout` + `AppBarLayout` don't work well with transparent statusbar, right?

I appreciated to you, so I'm so sorry to reject it :bow:

**Pull request closed without merging**